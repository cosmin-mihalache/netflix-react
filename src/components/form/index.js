import React from 'react';
import {
  Container,
  Base,
  Title,
  Submit,
  Text,
  SmallText,
  Input,
  Link,
  Error,
} from './styles/form';

export default function Form({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Form.Base = function FormBase({ children, ...restProps }) {
  return <Base {...restProps}>{children}</Base>;
};

Form.Title = function FormTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Form.Error = function FormError({ children, ...restProps }) {
  return <Error {...restProps}>{children}</Error>;
};

Form.Input = function FormInput({ children, ...restProps }) {
  return <Input {...restProps}>{children}</Input>;
};

Form.Submit = function FormSubmit({ children, ...restProps }) {
  return <Submit {...restProps}>{children}</Submit>;
};

Form.Link = function FormLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

Form.Text = function FormText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Form.SmallText = function FormSmallText({ children, ...restProps }) {
  return <SmallText {...restProps}>{children}</SmallText>;
};
