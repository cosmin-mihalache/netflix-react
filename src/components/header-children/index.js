import React from 'react';
import { Container, Title, SubTitle } from './styles/header-children';

export default function HeaderChild({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

HeaderChild.Title = function HeaderChildTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

HeaderChild.SubTitle = function HeaderChildSubTitle({
  children,
  ...restProps
}) {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};
