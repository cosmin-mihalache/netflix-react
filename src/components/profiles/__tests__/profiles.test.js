import React from 'react';
import { render } from '@testing-library/react';
import { Profiles } from '../../../components';
import { italic } from 'colorette';

describe('<Profiles />', () => {
  it('renders the <Profile /> with populated data', () => {
    const { container, getByText, getByTestId } = render(
      <Profiles>
        <Profiles.Title>Who's Watching?</Profiles.Title>
        <Profiles.List>
          <Profiles.User onClick={() => {}}>
            <Profiles.Picture
              src="/images/cosmin.png"
              data-testid="profile-picture"
            />
            <Profiles.Name>Cosmin Mihalache</Profiles.Name>
          </Profiles.User>
        </Profiles.List>
      </Profiles>
    );

    expect(getByText("Who's Watching?"));
    expect(getByTestId('profile-picture')).toBeTruthy();
    expect(getByText('Cosmin Mihalache')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the <Profiles /> with populated data but misc profile image', () => {
    const { container, getByText, getByTestId } = render(
      <Profiles>
        <Profiles.Title>Who's Watching?</Profiles.Title>
        <Profiles.List>
          <Profiles.User onClick={() => {}}>
            <Profiles.Picture data-testid="profile-picture-misc" />
            <Profiles.Name>Cosmin Mihalache</Profiles.Name>
          </Profiles.User>
        </Profiles.List>
      </Profiles>
    );

    expect(getByText("Who's Watching?"));
    expect(getByTestId('profile-picture-misc')).toBeTruthy();
    expect(getByText('Cosmin Mihalache')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
