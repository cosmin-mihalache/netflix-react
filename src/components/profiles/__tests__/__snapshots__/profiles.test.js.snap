// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profiles /> renders the <Profile /> with populated data 1`] = `
<div
  class="profiles__Container-ty3dj-0 hHBybT"
>
  <h1
    class="profiles__Title-ty3dj-1 hCMVSW"
  >
    Who's Watching?
  </h1>
  <ul
    class="profiles__List-ty3dj-2 ciDniQ"
  >
    <li
      class="profiles__User-ty3dj-5 iceCBY"
    >
      <img
        class="profiles__Picture-ty3dj-3 gTICYw"
        data-testid="profile-picture"
        src="/images/users//images/cosmin.png.png"
      />
      <p
        class="profiles__Name-ty3dj-4 hlGzaj"
      >
        Cosmin Mihalache
      </p>
    </li>
  </ul>
   
</div>
`;

exports[`<Profiles /> renders the <Profiles /> with populated data but misc profile image 1`] = `
<div
  class="profiles__Container-ty3dj-0 hHBybT"
>
  <h1
    class="profiles__Title-ty3dj-1 hCMVSW"
  >
    Who's Watching?
  </h1>
  <ul
    class="profiles__List-ty3dj-2 ciDniQ"
  >
    <li
      class="profiles__User-ty3dj-5 iceCBY"
    >
      <img
        class="profiles__Picture-ty3dj-3 gTICYw"
        data-testid="profile-picture-misc"
        src="/images/misc/loading.gif"
      />
      <p
        class="profiles__Name-ty3dj-4 hlGzaj"
      >
        Cosmin Mihalache
      </p>
    </li>
  </ul>
   
</div>
`;
